Error when updating user with groups

When a user has groups, updating the account cause a 500 error, with the stacktrace: 

```
java.lang.RuntimeException: java.lang.NullPointerException
	at fr.insee.sugoi.ldap.utils.mapper.GenericLdapMapper.mapObjectToLdapAttributes(GenericLdapMapper.java:123)
	at fr.insee.sugoi.ldap.utils.mapper.GenericLdapMapper.createMods(GenericLdapMapper.java:130)
	at fr.insee.sugoi.ldap.utils.mapper.UserLdapMapper.createMods(UserLdapMapper.java:45)
	at fr.insee.sugoi.store.ldap.LdapWriterStore.updateUser(LdapWriterStore.java:125)
	at fr.insee.sugoi.core.service.impl.UserServiceImpl.update(UserServiceImpl.java:57)
	at fr.insee.sugoi.services.controller.UserController.updateUsers(UserController.java:320)
        [....]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NullPointerException
	at fr.insee.sugoi.ldap.utils.mapper.GenericLdapMapper.lambda$createAttributesFromAttributeAnnotation$17(GenericLdapMapper.java:313)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at fr.insee.sugoi.ldap.utils.mapper.GenericLdapMapper.createAttributesFromAttributeAnnotation(GenericLdapMapper.java:315)
	at fr.insee.sugoi.ldap.utils.mapper.GenericLdapMapper.lambda$mapObjectToLdapAttributes$5(GenericLdapMapper.java:100)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at fr.insee.sugoi.ldap.utils.mapper.GenericLdapMapper.mapObjectToLdapAttributes(GenericLdapMapper.java:97)
	... 122 more
```

I thinks this is because the app_source key is not right here : https://github.com/InseeFrLab/sugoi-api/blob/2113db5673a71807f0d16547f11be4c554d12dc9/sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/mapper/GenericLdapMapper.java#L314

--- a/sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/mapper/GenericLdapMapper.java
+++ b/sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/mapper/GenericLdapMapper.java
@@ -92,7 +92,8 @@ public class GenericLdapMapper {
           .filter(
               ldapField ->
                   ldapField.getDeclaredAnnotationsByType(AttributeLdapName.class).length > 0
-                      && ldapField.getDeclaredAnnotationsByType(MapToAttribute.class).length > 0)
+                      && ldapField.getDeclaredAnnotationsByType(MapToAttribute.class).length > 0
+                      && !ldapField.getAnnotation(MapToAttribute.class).readonly())
           .forEach(
               ldapField -> {
                 List<Attribute> createdAttributeList =
@@ -297,7 +298,7 @@ public class GenericLdapMapper {
                                         .getAnnotation(LdapObjectClass.class)
                                         .rdnAttributeName(),
                                     group.getName(),
-                                    config.get("appli_source"))))
+                                    config.get("app_source"))))
                     .collect(Collectors.toList());
           case ADDRESS:
             List<Attribute> addressAttributeList = new ArrayList<>();

--- a/sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/mapper/properties/UserLdap.java
+++ b/sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/mapper/properties/UserLdap.java
@@ -134,6 +134,6 @@ public class UserLdap {
   private String habilitations;
 
   @AttributeLdapName("memberOf")
-  @MapToAttribute(value = "groups", type = ModelType.LIST_GROUP)
+  @MapToAttribute(value = "groups", type = ModelType.LIST_GROUP, readonly = true)
   private String groups;
 }

--- a/sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/mapper/properties/utils/MapToAttribute.java
+++ b/sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/mapper/properties/utils/MapToAttribute.java
@@ -28,4 +28,6 @@ public @interface MapToAttribute {
 
   /** Type d'objet */
   ModelType type() default ModelType.STRING;
+
+  boolean readonly() default false;
 }

--- a/sugoi-api-ldap-utils/src/test/java/fr/insee/sugoi/ldap/utils/mapper/ApplicationLdapMapperFromObjectTest.java
+++ b/sugoi-api-ldap-utils/src/test/java/fr/insee/sugoi/ldap/utils/mapper/ApplicationLdapMapperFromObjectTest.java
@@ -35,7 +35,7 @@ public class ApplicationLdapMapperFromObjectTest {
   public void setup() {
 
     Map<String, String> config = new HashMap<>();
-    config.put("appli_source", "ou=monappli,ou=Applications,o=insee,c=fr");
+    config.put("app_source", "ou=monappli,ou=Applications,o=insee,c=fr");
     config.put("address_source", "ou=address,o=insee,c=fr");
     applicationLdapMapper = new ApplicationLdapMapper(config);
 

